addons/mail/static/src/core/common/partner_compare.js

```diff
@@ -63,8 +63,8 @@ partnerCompareRegistry.add(
 partnerCompareRegistry.add(
     "mail.name",
     (p1, p2, { searchTerm }) => {
-        const cleanedName1 = cleanTerm(p1.name);
-        const cleanedName2 = cleanTerm(p2.name);
+        const cleanedName1 = normalize(p1.name);
```

Comment:
Just make sure it doesn't crash though. I don't think calling `startsWith` will work on non-string values.

For each case we need to determine if we need to fallback to empty string (as what did `cleanTerm`, which is the safe dist but conceptually a big shortcut) or handle undefined/falsy in a different way.